pkg/ext/delegate.go

```diff
@@ -0,0 +1,351 @@
+	ctx, err := s.makeContext(parentCtx)
+	if err != nil {
+		return nil, false, err
+	}
+
+	oldObj, err := s.store.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+
+	if deleteValidation != nil {
+		if err = deleteValidation(ctx, oldObj); err != nil {
+			return nil, false, err
+		}
+	}
+
+	err = s.store.Delete(ctx, name, options)
+	return oldObj, true, err
+}
+
+// Create implements [rest.Creater]
+//
+// createValidation is used to do some validation on the object before creating
+// it in the store. For example, running mutating/validating webhooks, though we're not using these yet.
+//
+//nolint:misspell
+func (s *delegate[T, TList]) Create(parentCtx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	ctx, err := s.makeContext(parentCtx)
+	if err != nil {
+		return nil, err
+	}
+
+	if createValidation != nil {
+		err := createValidation(ctx, obj)
+		if err != nil {
+			return obj, err
+		}
+	}
+
+	tObj, ok := obj.(T)
+	if !ok {
+		return nil, fmt.Errorf("object was of type %T, not of expected type %T", obj, s.t)
+	}
+
+	return s.store.Create(ctx, tObj, options)
+}
+
+// Update implements [rest.Updater]
+//
+// createValidation is used to do some validation on the object before creating
+// it in the store. For example, it will do an authorization check for "create"
+// verb if the object needs to be created.
+// See here for details: https://github.com/kubernetes/apiserver/blob/70ed6fdbea9eb37bd1d7558e90c20cfe888955e8/pkg/endpoints/handlers/update.go#L190-L201
+// Another example is running mutating/validating webhooks, though we're not using these yet.
+//
+// updateValidation is used to do some validation on the object before updating it in the store.
+// One example is running mutating/validating webhooks, though we're not using these yet.
+func (s *delegate[T, TList]) Update(
+	parentCtx context.Context,
+	name string,
+	objInfo rest.UpdatedObjectInfo,
+	createValidation rest.ValidateObjectFunc,
+	updateValidation rest.ValidateObjectUpdateFunc,
+	forceAllowCreate bool,
+	options *metav1.UpdateOptions,
+) (runtime.Object, bool, error) {
+	ctx, err := s.makeContext(parentCtx)
+	if err != nil {
+		return nil, false, err
+	}
+
+	oldObj, err := s.store.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, false, err
+		}
+
+		obj, err := objInfo.UpdatedObject(ctx, nil)
+		if err != nil {
+			return nil, false, err
+		}
+
+		if err = createValidation(ctx, obj); err != nil {
+			return nil, false, err
+		}
+
+		tObj, ok := obj.(T)
+		if !ok {
+			return nil, false, fmt.Errorf("object was of type %T, not of expected type %T", obj, s.t)
+		}
+
+		newObj, err := s.store.Create(ctx, tObj, &metav1.CreateOptions{})
+		if err != nil {
+			return nil, false, err
+		}
+		return newObj, true, err
```

Comment:
@MbolotSuse fixed this by returning true and I also added a test case for it to prevent regressions

Reply:
Ah I saw you left a comment for this, sorry for the useless ping.